src/components/app/UrlManager.js

```diff
@@ -38,7 +51,37 @@ type Props = ConnectedProps<OwnProps, StateProps, DispatchProps>;
  * This component manages the interaction with the window.history browser API.
  */
 class UrlManager extends React.PureComponent<Props> {
-  _updateState(firstRun: boolean) {
+  async _processInitialUrls() {
+    const {
+      startFetchingProfiles,
+      getProfilesFromRawUrl,
+      setupInitialUrlState,
+      urlSetupDone,
+    } = this.props;
+    let profile: Profile | null = null;
+    let error;
+
+    startFetchingProfiles();
+
+    try {
+      // Process the raw url and fetch the profile.
+      // $FlowFixMe Error introduced by upgrading to v0.96.0. See issue #1936.
+      profile = await getProfilesFromRawUrl(window.location);
+    } catch (err) {
+      error = err;
+    }
+
+    if (profile) {
+      setupInitialUrlState(window.location, profile);
+    } else if (error) {
+      // Just silently finish the url setup and return to home.
+      urlSetupDone();
+    } else {
+      throw new Error('This if branch should not be reached.');
```

Comment:
Maybe explain why it should not be reached, e.g. "An an unhandled case was reached during the initial processing of URLs."